File: tell-dont-ask-kata-ruby/models/OrderItem.rb
class OrderItem
    
    attr_accessor :product, :quantity, :taxed_amount, :tax

    def initialize(product: 'Unknown', quantity: 0, taxed_amount: 0.00, tax: 0.00)
        @product = product
        @quantity = quantity
        @taxed_amount = taxed_amount
        @tax = tax
    end
end

File: tell-dont-ask-kata-csharp/TellDontAskKata/domain/OrderItem.cs
public class OrderItem
{
    private Product product;
    private int quantity;
    private decimal taxedAmount;
    private decimal tax;

    public Product GetProduct()
    {
        return product;
    }

    public void SetProduct(Product product)
    {
        this.product = product;
    }

    public int GetQuantity()
    {
        return quantity;
    }

    public void SetQuantity(int quantity)
    {
        this.quantity = quantity;
    }

    public decimal GetTaxedAmount()
    {
        return taxedAmount;
    }

    public void SetTaxedAmount(decimal taxedAmount)
    {
        this.taxedAmount = taxedAmount;
    }

    public decimal GetTax()
    {
        return tax;
    }

    public void SetTax(decimal tax)
    {
        this.tax = tax;
    }
}